[AMDGPU] Generic lowering for rint and nearbyint

Acim-Maravic opened the pull request:
The are three different rounding intrinsics, that are brought down to same instruction.

Acim-Maravic commented:
Continuation of: https://reviews.llvm.org/D159271

arsenm commented on llvm/include/llvm/CodeGen/GlobalISel/MachineIRBuilder.h:
```diff
@@ -1888,6 +1888,14 @@ class MachineIRBuilder {
 
+  /// Build and insert \p Dst = G_INTRINSIC_ROUNDEVEN \p Src0, \p Src1
+  MachineInstrBuilder
+  buildintrinsicRoundeven(const DstOp &Dst, const SrcOp &Src0,
```
Capitalization is off. I'd also probably just drop the intrinsic part from the name (we should also drop it from the opcode name)

Acim-Maravic committed:
[AMDGPU] Generic lowering for rint and nearbyint

There are three different rounding intrinsics, that are brought down to same instruction.
